src/modules/ekf2/EKF/ekf.cpp

```diff
@@ -115,6 +115,14 @@ bool Ekf::update()
 
 		// run EKF-GSF yaw estimator
 		runYawEKFGSF();
+
+
+		// fix gross errors in the covariance matrix and ensure rows and
+		// columns for un-used states are zero
+		if (_covariance_updated) {
+			fixCovarianceErrors(true);
```

Comment:
TODO: consider running this at the beginning of the next cycle instead? Then in the output data (estimator_status, etc) we might have some visibility in the necessity of these checks/fixes.

Reply:
PR updated to test.